test/components/structures/TimelinePanel-test.tsx

```diff
@@ -58,7 +61,7 @@ const newReceipt = (eventId: string, userId: string, readTs: number, fullyReadTs
 };
 
 const getProps = (room: Room, events: MatrixEvent[]): TimelinePanel["props"] => {
-    const timelineSet = { room: room as Room } as EventTimelineSet;
+    const timelineSet = room.getUnfilteredTimelineSet?.() ?? ({ room: room as Room } as EventTimelineSet);
```

Comment:
is this hoop-jumping because we sometimes use a mocked-out Room rather than a real one with a real `getUnfilteredTimelineSet` ?

Is it possible for `getUnfilteredTimelineSet()` to return undefined and for us to use the mocked-out EventTimelineSet even when we have a real room? or does that happen only if there is no `getUnfilteredTimelineSet` method?

Reply:
isn't `room as Room` redundant?

```suggestion
    const timelineSet = room.getUnfilteredTimelineSet?.() ?? ({ room } as EventTimelineSet);
```

Reply:
yes, and I have opted to revert the changes here and not use `getProps` in the first place.